e2e/helpers/environment/EnvironmentManager.ts

```diff
@@ -69,20 +69,61 @@ export class EnvironmentManager {
         }
     }
 
+    /**
+     * Clean up leftover resources from previous test runs
+     * This should be called at the start of globalSetup to ensure a clean slate,
+     * especially after interrupted test runs (e.g. via ctrl+c)
+     *
+     * 1. Remove all leftover Ghost containers
+     * 2. Clean up leftover test databases (if MySQL is running)
+     * 3. Delete the MySQL snapshot (if MySQL is running)
+     * 4. Recreate the ghost_testing database (if MySQL is running)
+     * 5. Truncate Tinybird analytics_events datasource (if Tinybird is running)
+     *
+     * Note: Docker compose services are left running for reuse across test runs
+     */
+    private async cleanupLeftoverResources(): Promise<void> {
+        try {
+            logging.info('Cleaning up leftover resources from previous test runs...');
+
+            // Clean up leftover Ghost containers
```

Comment:
`nit` function and intentions are clear, comments seem redundant within function